### PR TITLE
ADT_Assignment_22_Done_15_10Mins

### PR DESCRIPTION
**Pull request by AditiGJUN20**
**Assignment 22**
**1. PalindromeNumber.java**
Time: 10Mins
Screen Shot 2020-07-01 at 1 10 00 AM

**2. PalindromeString.java **
Time: 15Mins
Screen Shot 2020-07-01 at 1 23 21 AM

### CODE REVIEW

**Review by AmitaRJun20 — APPROVED** [review]
Codes are perfectly written as per expectation... Good job...

File: src/aditiGJUN20/PalindromeNumber.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+public class PalindromeNumber {
+
+	// Method to check number is palindrome or not.
+	void checkPalindromeNumber(int num) {
+		int reverseNum = 0, inputNum = num;
+		System.out.println("\nInput Number = " + inputNum);
+		while (num > 0) {
+			reverseNum = reverseNum * 10 + (num % 10);
+			num = num / 10;
```
Comment (KrishnaKTechnocredits) [inline src/aditiGJUN20/PalindromeNumber.java:22]: we can write assignment operation in while loop. Just FYI.

while((number = number/10)>0)

File: src/aditiGJUN20/PalindromeString.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+package aditiGJUN20;
+
+public class PalindromeString {
+	boolean isPalindromString(String input) {
+		System.out.println("Input String: "+input);
+		String rev="";
+		for ( int index = input.length() - 1; index >= 0; index-- )
+        rev = rev + input.charAt(index); 
+		
+		if(input.equalsIgnoreCase(rev))
```
Comment (KrishnaKTechnocredits) [inline src/aditiGJUN20/PalindromeString.java:17]: Better to use equal method. If interviewer asked to checking it without case sensitivity you should equalsIgnoreCase

**Review by KrishnaKTechnocredits — APPROVED** [review]
Code looks good to me. Please take a look at comments.

File: src/aditiGJUN20/PalindromeString.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * Output: Given string technocredits is not palindrome.
+*/
+package aditiGJUN20;
+
+public class PalindromeString {
+	boolean isPalindromString(String input) {
+		System.out.println("Input String: "+input);
+		String rev="";
+		for ( int index = input.length() - 1; index >= 0; index-- )
+        rev = rev + input.charAt(index);
```
Comment (KrishnaKTechnocredits) [inline src/aditiGJUN20/PalindromeString.java:15]: you can compare string from both end without having rev [extra] variable.

**Review by KrishnaKTechnocredits — APPROVED** [review]

File: src/aditiGJUN20/PalindromeNumber.java (hunk 1 above)
Comment (AditiGJUN20) [reply src/aditiGJUN20/PalindromeNumber.java:22]: This is new.  i was aware about for loop. But not knowing that for while loop also we can create similar behavior.
Thank you Krishna.

File: src/aditiGJUN20/PalindromeString.java (hunk 2 above)
Comment (AditiGJUN20) [reply src/aditiGJUN20/PalindromeString.java:17]: I'll keep that in mind.

**Pull request merged**